ee/vellum_ee/workflows/display/nodes/vellum/inline_subworkflow_node.py

```diff
@@ -60,12 +61,25 @@ def _generate_node_and_workflow_inputs(
         node: Type[InlineSubworkflowNode],
         display_context: WorkflowDisplayContext,
     ) -> Tuple[List[NodeInput], List[VellumVariable]]:
+        subworkflow = raise_if_descriptor(node.subworkflow)
+        subworkflow_inputs_class = subworkflow.get_inputs_class()
         subworkflow_inputs = raise_if_descriptor(node.subworkflow_inputs)
-        subworkflow_entries = (
-            [(variable_name, variable_value) for variable_name, variable_value in subworkflow_inputs.items()]
-            if isinstance(subworkflow_inputs, dict)
-            else [(variable_ref.name, variable_value) for variable_ref, variable_value in subworkflow_inputs]
-        )
+
+        if isinstance(subworkflow_inputs, BaseInputs):
```

Comment:
spent some cycles trying to consolidate this with `_compile_subworkflow_inputs`, but one operates on the instance and one on the class, and it wasn't worth converting the former to the latter because of nested node resolution.